### PR TITLE
effective user with non-ascii password

### PR DESCRIPTION
**Pull request by pondrejk**
### Problem Statement
covers bz# 2258968

### Solution
testing rex with effective user with non-ascii password

### CODE REVIEW

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8]

**Comment by pondrejk** [comment]
local result against patched stream
```
pytest tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8]
======================================== test session starts ========================================

collected 1 item                                                                                    

tests/foreman/cli/test_remoteexecution.py .                                                   [100%]
```

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8]

**Comment by pondrejk** [comment]
moving to draft until the changes appear in stream

File: tests/foreman/cli/test_remoteexecution.py [hunk 1]
```diff
@@ -161,21 +161,24 @@ def test_positive_run_job_effective_user(self, rex_contenthost, module_target_sa
...
             effective user identity on host
 
         :parametrized: yes
+
+        :customerscenario: true
         """
         client = rex_contenthost
         # create a user on client via remote job
         username = gen_string('alpha')
+        password = gen_string('cjk')
```
[the remark this answers is not in the document]
Comment (pondrejk) [reply tests/foreman/cli/test_remoteexecution.py:176]: the covered bz 2258968 asks for using something nonASCII as a password...

File: tests/foreman/cli/test_remoteexecution.py [hunk 2]
```diff
@@ -161,21 +161,24 @@ def test_positive_run_job_effective_user(self, rex_contenthost, module_target_sa
...
         filename = gen_string('alpha')
         make_user_job = module_target_sat.cli_factory.job_invocation(
             {
                 'job-template': 'Run Command - Script Default',
-                'inputs': f"command=useradd -m {username}",
+                'inputs': f"command=useradd -m {username} -p {password} -G wheel",
```
Comment (lhellebr) [inline tests/foreman/cli/test_remoteexecution.py:181]: Given running a job worked previously without the password specified (Using sudo?), are we positive that the password is really being used in this new version?

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8]

**Comment by pondrejk** [comment]
Awaiting stream snap 51, at theat point we should get passing prt

File: tests/foreman/cli/test_remoteexecution.py [hunk 3]
```diff
@@ -187,6 +190,7 @@ def test_positive_run_job_effective_user(self, rex_contenthost, module_target_sa
                 'inputs': f"command=touch /home/{username}/{filename}",
                 'search-query': f"name ~ {client.hostname}",
                 'effective-user': f'{username}',
+                'effective-user-password': f'{password}',
```
[the remark this answers is not in the document]
Comment (pondrejk) [reply tests/foreman/cli/test_remoteexecution.py:199]: Hi, I believe I'm doing that few lines bellow for user, I added a check for group as well.

File: tests/foreman/cli/test_remoteexecution.py (hunk 2 above)
Comment (pondrejk) [reply tests/foreman/cli/test_remoteexecution.py]: I made some changes to reduce the possibility of secret sudo, but the only way I could find to be 100% sure is a negative assert -- running rex without pwd specified, so I added that as well

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8]

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6217
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8] --external-logging
Test Result : ================== 1 failed, 4 warnings in 1091.27s (0:18:11) ==================
```

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8]

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6224
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8] --external-logging
Test Result : ================== 1 passed, 4 warnings in 1130.62s (0:18:50) ==================
```

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8]

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6522
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8] --external-logging
Test Result : ================= 1 passed, 13 warnings in 1111.73s (0:18:31) ==================
```

**Comment by pondrejk** [comment]
prt passed but some urelated problem made robotelo-runner status red, discusedd in automation channel. Retriggering prt in hope that all will be well again

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8]

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 6555
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_remoteexecution.py::TestRemoteExecution::test_positive_run_job_effective_user[rhel8] --external-logging
Test Result : ================= 1 passed, 13 warnings in 1095.98s (0:18:15) ==================
```

**Review by jameerpathan111 — APPROVED** [review]

**Pull request merged**